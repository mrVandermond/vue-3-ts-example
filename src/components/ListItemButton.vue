<template>
  <div
    :class="{
      'is-open': props.isOpen,
    }"
    class="list-item-button"
  />
</template>

<script lang="ts" setup>
const props = defineProps<{
  isOpen: boolean;
}>();
</script>

<style lang="scss">
.list-item-button {
  width: 32px;
  height: 32px;
  border-radius: 50%;
  position: relative;
  background-color: $cyan;

  &:before,
  &:after {
    content: '';
    position: absolute;
    background-color: $white;
  }

  &:before {
    width: 2px;
    height: 14px;
    top: calc(50% - 7px);
    left: calc(50% - 1px);
  }

  &:after {
    width: 14px;
    height: 2px;
    top: calc(50% - 1px);
    left: calc(50% - 7px);
  }

  &.is-open:before {
    display: none;
  }
}
</style>
