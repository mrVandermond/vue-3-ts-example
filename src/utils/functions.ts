export function getItemIndexBy<T>(array: T[], item: T, field: keyof T): number {
  return array.findIndex((temp) => temp[field] === item[field]);
}
