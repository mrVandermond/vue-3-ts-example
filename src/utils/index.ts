export * as utlFuncs from './functions';
export * as utlCst from './constants';
