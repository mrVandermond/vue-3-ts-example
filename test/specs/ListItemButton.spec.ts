import ListItemButton from '@/components/ListItemButton.vue';

import type { TWrapperFactoryReturnType } from '../utils';
import { wrapperFactory } from '../utils';

describe('ListItemButton', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const getWrapper = (props: any): TWrapperFactoryReturnType => wrapperFactory(ListItemButton, {
    props,
  });

  it('Рендер в закрытом состоянии', () => {
    const wrapper = getWrapper({
      isOpen: false,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Рендер в открытом состоянии', () => {
    const wrapper = getWrapper({
      isOpen: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
