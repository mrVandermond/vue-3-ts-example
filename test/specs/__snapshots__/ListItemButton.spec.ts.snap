// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemButton Рендер в закрытом состоянии 1`] = `
<div class="list-item-button">
</div>
`;

exports[`ListItemButton Рендер в открытом состоянии 1`] = `
<div class="is-open list-item-button">
</div>
`;
